nni/algorithms/compression/pytorch/pruning/iterative_pruner.py

```diff
@@ -120,14 +119,16 @@ class AGPPruner(IterativePruner):
 
     def __init__(self, model, config_list, optimizer, trainer, criterion,
                  num_iterations=10, epochs_per_iteration=1, pruning_algorithm='level'):
-        super().__init__(model, config_list, optimizer=optimizer, trainer=trainer, criterion=criterion,
+        self.freq = epochs_per_iteration
+        self.end_epoch = epochs_per_iteration * num_iterations
```

Comment:
why move these two lines to the top?

Reply:
move all constant assignments before init, because we  call `reconfig()` in `__init__`, and these constant may be used in `reconfig()`.

Reply:
got it